api/controllers/user.go

```diff
@@ -0,0 +1,38 @@
+package controllers
+
+import (
+  "net/http"
+  "strings"
+  "github.com/gin-gonic/gin"
+  "github.com/charmbracelet/log"
+
+  "github.com/Salz-Team/salz/api/db"
+)
+
+func GetUser(c *gin.Context, dbHandler db.ApiDBHandler, authHandler db.AuthDBHandler) {
+  // Get the user from the database
+  
+  authTokenHeader := c.GetHeader("Authorization")
+  authToken := strings.TrimSpace(strings.Replace(authTokenHeader, "Bearer", "", 1))
+
+  token, err := authHandler.GetToken(authToken)
+  if err != nil {
+    log.Error("Unable to get token", "error", err)
+    c.AbortWithStatus(http.StatusInternalServerError)
+    return
+  }
```

Comment:
Is this what we want? Or just current behaviour until we decide what's better? I'd think that if you've somehow passed an auth guard (well, this assumes that this function is put behind an auth guard, which it is in `main.go`), and somehow your token's gone or expired, then return a 401?

Reply:
mostly placeholder behaviour. This call could be avoided if we saved the user's salz-id in the context after validating the token in the auth middleware, instead of having to correlate the token and the salz-id for every protected endpoint.

Reply:
(that should be its own ticket) https://github.com/Salz-Team/salz/issues/63